### PR TITLE
Munemo Improvements

### PR DESCRIPTION
**Pull request by directionless**
Several munemo improvements. May not be ideal go. 

1. Test cases 
2. Repairs handling of negative numbers
3. Adds an decode function to reverse

### CODE REVIEW

File: munemo/munemo.go [hunk 1]
```diff
@@ -14,14 +15,23 @@ func Munemo(id int) string {
...
+	m.decode(s)
+	return m.int() * m.sign
+}
+
 type munemo struct {
 	negativeSymbol string
 	symbols        []string
 	buffer         *bytes.Buffer
+	number         int
+	symbol_values  map[string]int
```
Comment (groob) [inline munemo/munemo.go]: For Go code, use camel case.

File: munemo/munemo_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package munemo
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMunemo(t *testing.T) {
```
Comment (groob) [inline munemo/munemo_test.go]: nice!

File: munemo/munemo.go [hunk 3]
```diff
@@ -14,14 +15,23 @@ func Munemo(id int) string {
 	return m.string()
 }
 
+func UnMunemo(s string) int {
```
Comment (groob) [inline munemo/munemo.go]: If this can fail, lets add an error return.

File: munemo/munemo.go [hunk 4]
```diff
@@ -44,19 +54,63 @@ func newMunemo() *munemo {
...
 	return m.buffer.String()
 }
 
+func (m *munemo) int() int {
+	return m.number
+}
+
+func (m *munemo) decode(s string) {
+	arr := []byte(s)
+	m.buffer.Write(arr)
```
Comment (groob) [inline munemo/munemo.go]: this buffer is unused during the conversion. no need to write to it.

File: munemo/munemo.go [hunk 5]
```diff
@@ -44,19 +54,63 @@ func newMunemo() *munemo {
...
 func (m *munemo) string() string {
 	return m.buffer.String()
 }
 
+func (m *munemo) int() int {
+	return m.number
+}
+
+func (m *munemo) decode(s string) {
+	arr := []byte(s)
```
Comment (groob) [inline munemo/munemo.go]: No need to convert to the byte slice. All the operations below would still work on the raw string. 
Ex: `s = s[2:len(s)]`

File: munemo/munemo.go [hunk 6]
```diff
@@ -44,19 +54,63 @@ func newMunemo() *munemo {
...
+	// negative if the first two bytes match the negative symbol
+	if string(arr[0:2]) == m.negativeSymbol {
+		m.sign = -1
+		arr = arr[2:len(arr)]
+	}
+
+	// As long as there are characters, parse them
+	// Read the first syllable, interpret, remove.
+	for {
+		if len(arr) == 0 {
```
Comment (groob) [inline munemo/munemo.go]: `if s == ""` when working with a string.

File: munemo/munemo.go (hunk 3 above)
Comment (directionless) [reply munemo/munemo.go]: yeah. I wasn't sure of the idiom. This code seems mostly OO oriented. Creates an object, then calls methods on said object. No real returns, per se. I'll try to mock up something

File: munemo/munemo.go (hunk 4 above)
Comment (directionless) [reply munemo/munemo.go]: Yeah. this code is a weird mix of functional and OO. I left that there as a start towards making it more OO'ish. But lemme rip it out. Maybe we should go more functional

**Pull request merged**